src/tools/wasm2js.cpp

```diff
@@ -55,28 +55,43 @@ int main(int argc, const char *argv[]) {
   options.parse(argc, argv);
   if (options.debug) builderFlags.debug = true;
 
-  auto input(
-      read_file<std::vector<char>>(options.extra["infile"], Flags::Text, options.debug ? Flags::Debug : Flags::Release));
-
   Element* root;
   Module wasm;
   Ref asmjs;
 
   try {
-    if (options.debug) std::cerr << "s-parsing..." << std::endl;
-    SExpressionParser parser(input.data());
-    root = parser.root;
+    // If the input filename ends in `.wasm`, then parse it in binary form
+    auto &input = options.extra["infile"];
+    std::string suffix(".wasm");
+    if (input.size() >= suffix.size() &&
+        input.compare(input.size() - suffix.size(), suffix.size(), suffix) == 0) {
+      ModuleReader reader;
+      reader.setDebug(options.debug);
+      reader.read(input, wasm, "");
+
+      if (options.debug) std::cerr << "asming..." << std::endl;
+      Wasm2JSBuilder wasm2js(builderFlags);
+      asmjs = wasm2js.processWasm(&wasm);
+
+    // Otherwise assume it's a `*.wast` file and go from there
```

Comment:
maybe add a comment about why we can't just use ModuleReader everywhere, adding "because we need to process the assertions too"?